### PR TITLE
#170947448 comment on  a travel requests

### PR DESCRIPTION
**Pull request by blaise82**
### What does this PR do?

- create comments table
- create comments route
- create comment controller
- write test
### How should this be manually tested?

- clone this repo using `git clone https://github.com/andela/octopus-bn-backend`
- run `cd octopus-bn-backend` to move to the directory
- run `npm install` to install all required dependencies
- run `npm run dev` to start the development server
- open postman POST localhost:3000/api/v1/requests/comment?requestId={request id for the request you want to comment on}
- add the following body data example 
```
{
    "comment": "string",
  }
```

### Stories associated with PR

[#170947448](https://www.pivotaltracker.com/story/show/170947448)

### Screenshots if any
Screen Shot 2020-03-05 at 10 36 39

[finishes #170947448]

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -39,7 +43,4 @@
       }
     }
   },
-  "parserOptions": {
-    "sourceType": "module"
-}
-}
+}
```
Comment (hakoemmy) [inline .eslintrc.json]:
```suggestion
}

```

File: .hound.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 eslint:
   enabled: true
   config_file: .eslintrc
-  
\ No newline at end of file
+  ignore_file: .eslintignore
```
Comment (hakoemmy) [inline .hound.yml:4]:
```suggestion
  ignore_file: .eslintignore
  
```

File: src/database/migrations/20200303093352-create-comments.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    createdAt: {
+      allowNull: false,
+      type: Sequelize.DATE
+    },
+    updatedAt: {
+      allowNull: false,
+      type: Sequelize.DATE
+    }
+  }),
+  // eslint-disable-next-line no-unused-vars
```
Comment (hakoemmy) [inline src/database/migrations/20200303093352-create-comments.js]: @blaise82 Add the rule to ```.eslintIgnore`` or try to fix it like so

File: src/middlewares/validateRejectTrip.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ export const isTripExist = async (req, res, next) => {
       tripId
     }
   } = req;
-  const trip = await travelRequests.findOne({ where: { requestId: tripId } });
+  const trip = await travelRequests.findOne({ where: { id: tripId } });
```
Comment (hakoemmy) [inline src/middlewares/validateRejectTrip.js]: @blaise82 Why did you change ```requestId``` to ``id``

Comment (blaise82) [reply src/middlewares/validateRejectTrip.js]: it was to make sure that when someone fetches requests we can also get comments made to that request 

using table relationships
our tables had no relationships and sequelize  looks for the id to refer to

File: src/database/migrations/20200303093352-create-comments.js (hunk 3 above)
Comment (blaise82) [reply src/database/migrations/20200303093352-create-comments.js]: I can't ignore that rule it helps sometimes
but let me just add specific files to eslintIgnore

Comment (hakoemmy) [reply src/database/migrations/20200303093352-create-comments.js]: @blaise82 Sure thing

File: src/middlewares/validateRejectTrip.js (hunk 4 above)
Comment (hakoemmy) [reply src/middlewares/validateRejectTrip.js]: @blaise82 Sure thing, I understand

**Review by hakoemmy — APPROVED** [review]

**Review by RusimbiPatrick — APPROVED** [review]

**Review by aaronsekisambu — APPROVED** [review]
